Tests/TensorFlowTests/OperatorTests/MathTests.swift

```diff
@@ -65,6 +65,14 @@ final class MathOperatorTests: XCTestCase {
         assertEqual(y, log(1 + x), accuracy: 0.0001)
     }
 
+    func testCosineSimilarity() {
+        let x = Tensor<Float>([1, 2, 3, 4, 5, 6, 7, 8])
+        let y = Tensor<Float>([0.5, 1, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0])
+        let z = cosineSimilarity(x, y)
+        let output: Float = 1.0
+        XCTAssertEqual(z, Tensor(output))
+    }
+
```

Comment:
Could you please also add a test for cosine distance?

Reply:
@saeta, actually cosine distance just returns 1 - cosineSimilarity, so in effect we would be testing the same function. To avoid redundancy I didn't add a test

Reply:
Even though the function is implemented as a trivial wrapper, it's good practice to assume the implementation can change in the future for whatever reasons and have tests be implementation-agnostic.